### PR TITLE
added custom keyword search  feature

### PR DESCRIPTION
**Pull request by nishabalamurugan**

### CODE REVIEW

File: xgitguard/custom keyword search/public_keyword_search.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+            logger.error(f"Process Error: {e}")
+        if new_results_per_query >= 0:
+            logger.info(
+                f"Total: {total_search_pairs} "
+                + f"Processed: {total_processed_search} "
+            )
+    logger.info(f"Current Total Processed Search: {total_processed_search}")
+    logger.info(f"Current Total Detections Write: {total_detection_writes}")
+
+    return True
```
Comment (sai100) [inline xgitguard/custom keyword search/public_keyword_search.py:297]: Please keep the changes similar to enterprise keyword search module.

File: xgitguard/common/github_calls.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def run_github_search(self, search_query, extension, org=[], repo=[]):
                 )
                 sys.exit(1)
 
-        if not extension and extension == "others":
+        if not extension and extension == "others" or len(extension) == 0:
```
Comment (sai100) [inline xgitguard/common/github_calls.py:74]: Please  replaced and with or

File: xgitguard/custom keyword search/public_keyword_search.py [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+    new_urls_list, new_hashed_urls = [], []
+    global file_prefix
+    # Get the Already predicted hashed url list if present
+    try:
+        # for Reading training Data only one time
+        if configs.hashed_urls:
+            pass
+    except:
+        configs.read_hashed_url(
+            file_name=file_prefix + "public_hashed_url_custom_keywords.csv"
```
Comment (sai100) [inline xgitguard/custom keyword search/public_keyword_search.py:131]: Please rename file name, make it similar to enterprise file name

**Review by sai100 — CHANGES_REQUESTED** [review]
Please make the changes as mentioned in review comments

File: xgitguard/custom keyword search/public_keyword_search.py (hunk 1 above)
Comment (nishabalamurugan) [reply xgitguard/custom keyword search/public_keyword_search.py]: updated

File: xgitguard/common/github_calls.py (hunk 2 above)
Comment (nishabalamurugan) [reply xgitguard/common/github_calls.py]: updated

**Review by sai100 — CHANGES_REQUESTED** [review]
please make the changes as requested in comments

File: xgitguard/custom keyword search/enterprise_keyword_search.py [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+    detection_writes_per_query = 0
+    new_results_per_query = 0
+    detections_per_query = 0
+    new_hashed_urls = []
+    global file_prefix
+
+    url_list, org_url_list = [], []
+
+    hashed_urls_file = os.path.join(
+        configs.output_dir, file_prefix + "enterprise_custom_keyword_hashed_url.csv"
```
[the remark this answers is not in the document]
Comment (nishabalamurugan) [reply xgitguard/custom keyword search/enterprise_keyword_search.py]: updated

File: README.md [hunk 5]
```diff
@@ -726,6 +722,100 @@ Traverse into the "ml_training" folder
...
+
+##### Command-Line Arguments for Enterprsie keyword Scanner
+
+```
+Run usage:
+enterprise_keyword_search.py [-h] [-e Enterprise Keywords]  [-o org_name] [-r repo_name] [-l Logger Level] [-c Console Logging]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -e Enterprise Keywords, --enterprise_keywords Enterprise Keywords
```
Comment (dinpraka) [inline README.md:766]: Add this as optional to search with keywords in cmd line

Comment (nishabalamurugan) [reply README.md:766]: updated

**Review by dinpraka — APPROVED** [review]

**Review by sai100 — APPROVED** [review]
Approved

**Pull request merged**